vendor: github.com/moby/buildkit v0.8.3-3-g244e8cde

thaJeztah opened the pull request:
### vendor: github.com/moby/buildkit v0.8.3-3-g244e8cde

full diff: https://github.com/moby/buildkit/compare/v0.8.3...v0.8.3-3-g244e8cde

- Transform relative mountpoints for exec mounts in the executor
- Add test for handling relative mountpoints

### vendor: github.com/moby/buildkit v0.8.3

full diff: https://github.com/moby/buildkit/compare/v0.8.2...v0.8.3

- vendor containerd (required for rootless overlayfs on kernel 5.11)
    - not included to avoid depending on a fork
- Add retry on image push 5xx errors
- contenthash: include basename in content checksum for wildcards
- Fix missing mounts in execOp cache map
- Add regression test for run cache not considering mounts
- Add hack to preserve Dockerfile RUN cache compatibility after mount cache bugfix

**- Description for the changelog**
<!--
Write a short (one line) summary that describes the changes in this
pull request for inclusion in the changelog:
-->


**- A picture of a cute animal (not mandatory but encouraged)**

thaJeztah committed:
vendor: github.com/moby/buildkit v0.8.3

full diff: https://github.com/moby/buildkit/compare/v0.8.2...v0.8.3

- vendor containerd (required for rootless overlayfs on kernel 5.11)
    - not included to avoid depending on a fork
- Add retry on image push 5xx errors
- contenthash: include basename in content checksum for wildcards
- Fix missing mounts in execOp cache map
- Add regression test for run cache not considering mounts
- Add hack to preserve Dockerfile RUN cache compatibility after mount cache bugfix

Signed-off-by: Sebastiaan van Stijn <github@gone.nl>

thaJeztah commented:
@tonistiigi @AkihiroSuda PTAL

I didn't take the containerd vendoring changes to avoid having to depend on a fork; is there a way we can use (e.g.) a tagged version of containerd with the needed fixes?

thaJeztah commented on vendor/github.com/moby/buildkit/go.mod:
```diff
@@ -71,6 +72,11 @@ require (
+	// containerd: Forked from 0edc412565dcc6e3d6125ff9e4b009ad4b89c638 (20201117) with:
+	// - `Adjust overlay tests to expect "index=off"`        (#4719, for ease of cherry-picking #5076)
+	// - `overlay: support "userxattr" option (kernel 5.11)` (#5076)
+	// - `docker: avoid concurrent map access panic`         (#4855)
```
Is this related to https://github.com/moby/moby/issues/42422 ?

tonistiigi replied:
no

AkihiroSuda commented:
For v21.xx, can we rather vendor the master branch?

thaJeztah commented:
I think it's ok to update after that; thought we'd take this one (possibly cherry-pick for 20.10)

I need to look at updating runc etc (had some WIP pr's for that) for master

tonistiigi commented:
Yeah, we should pick v0.8.3 + https://github.com/moby/buildkit/pull/2137 directly to 20.10 .

thaJeztah committed:
vendor: github.com/moby/buildkit v0.8.3-3-g244e8cde

full diff: https://github.com/moby/buildkit/compare/v0.8.3...v0.8.3-3-g244e8cde

- Transform relative mountpoints for exec mounts in the executor
- Add test for handling relative mountpoints

Signed-off-by: Sebastiaan van Stijn <github@gone.nl>

thaJeztah commented:
added https://github.com/moby/buildkit/pull/2137

and opened https://github.com/moby/moby/pull/42448 for 20.10
